diff -up plasma6-systemsettings-6.3.3/app/kcmmetadatahelpers.h.omv~ plasma6-systemsettings-6.3.3/app/kcmmetadatahelpers.h
--- plasma6-systemsettings-6.3.3/app/kcmmetadatahelpers.h.omv~	2025-03-14 02:08:56.117329189 +0100
+++ plasma6-systemsettings-6.3.3/app/kcmmetadatahelpers.h	2025-03-14 02:09:51.053646943 +0100
@@ -75,7 +75,7 @@ inline QList<KPluginMetaData> findKCMsMe
 
     auto filter = [](const KPluginMetaData &data) {
         const auto supportedPlatforms = data.value(QStringLiteral("X-KDE-OnlyShowOnQtPlatforms"), QStringList());
-        return supportedPlatforms.isEmpty() || supportedPlatforms.contains(qGuiApp->platformName());
+        return supportedPlatforms.isEmpty() || supportedPlatforms.contains(qGuiApp->platformName()) || (qGuiApp->platformName().startsWith(QStringLiteral("wayland")) && supportedPlatforms.contains(QStringLiteral("wayland")));
     };
 
     // We need the exist calls because otherwise the trader language aborts if the property doesn't exist and the second part of the or is not evaluated
